invenio/modules/indexer/testsuite/test_indexer_engine.py

```diff
@@ -19,65 +19,72 @@
 
 """Unit tests for the indexing engine."""
 
-__revision__ = \
-    "$Id$"
-
 from invenio.base.wrappers import lazy_import
-from invenio.testsuite import make_test_suite, run_test_suite, InvenioTestCase
+from invenio.testsuite import InvenioTestCase, make_test_suite, run_test_suite
+
 bibindex_engine = lazy_import('invenio.legacy.bibindex.engine')
-load_tokenizers = lazy_import('invenio.legacy.bibindex.engine_utils.load_tokenizers')
+load_tokenizers = lazy_import(
+    'invenio.legacy.bibindex.engine_utils.load_tokenizers')
 list_union = lazy_import('invenio.legacy.bibindex.engine_utils.list_union')
-get_values_recursively = lazy_import('invenio.legacy.bibindex.engine_utils.get_values_recursively')
+get_values_recursively = lazy_import(
+    'invenio.legacy.bibindex.engine_utils.get_values_recursively')
 
 
 class TestListSetOperations(InvenioTestCase):
+
     """Tests for list set operations."""
 
     def test_list_union(self):
-        """bibindex engine utils - list union"""
+        """bibindex engine utils - list union."""
         self.assertEqual([1, 2, 3, 4],
                          list_union([1, 2, 3],
                                     [1, 3, 4]))
 
     def test_list_unique(self):
-        """bibindex engine - list unique"""
+        """bibindex engine - list unique."""
         self.assertEqual([1, 2, 3],
                          bibindex_engine.list_unique([1, 2, 3, 3, 1, 2]))
 
 
-
 class TestWashIndexTerm(InvenioTestCase):
+
     """Tests for washing index terms, useful for both searching and indexing."""
 
     def test_wash_index_term_short(self):
-        """bibindex engine - wash index term, short word"""
+        """bibindex engine - wash index term, short word."""
         self.assertEqual("ellis",
                          bibindex_engine.wash_index_term("ellis"))
 
     def test_wash_index_term_long(self):
-        """bibindex engine - wash index term, long word"""
-        self.assertEqual(50*"e",
-                         bibindex_engine.wash_index_term(1234*"e"))
+        """bibindex engine - wash index term, long word."""
+        self.assertEqual(50 * "e",
+                         bibindex_engine.wash_index_term(1234 * "e"))
 
     def test_wash_index_term_case(self):
-        """bibindex engine - wash index term, lower the case"""
+        """bibindex engine - wash index term, lower the case."""
         self.assertEqual("ellis",
                          bibindex_engine.wash_index_term("Ellis"))
 
     def test_wash_index_term_unicode(self):
-        """bibindex engine - wash index term, unicode"""
+        """bibindex engine - wash index term, unicode."""
         self.assertEqual("ελληνικό αλφάβητο",
-          bibindex_engine.wash_index_term("Ελληνικό αλφάβητο"))
+                         bibindex_engine.wash_index_term("Ελληνικό αλφάβητο"))
+
+    def test_wash_index_term_unicode_for_real(self):
+        """bibindex engine - wash index term, unicode string."""
+        self.assertEqual("ellis",
```

Comment:
Here I added a new test, as the washer did not like to be given Unicode suddenly :boom: